File: 108. Convert Sorted Array to Binary Search Tree/108. Convert Sorted Array to Binary Search Tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public TreeNode SortedArrayToBST(int[] nums) {
        return BuildBST(nums, 0, nums.Length - 1);
    }
    
    private TreeNode BuildBST(int[] nums, int start, int end) {
        if (start > end) return null;
        
        int mid = start + (end - start) / 2;
        TreeNode node = new TreeNode(nums[mid]);
        node.left = BuildBST(nums, start, mid - 1);
        node.right = BuildBST(nums, mid + 1, end);
        
        return node;
    }
}

File: 108. Convert Sorted Array to Binary Search Tree/108. Convert Sorted Array to Binary Search Tree.dart
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *   int val;
 *   TreeNode? left;
 *   TreeNode? right;
 *   TreeNode([this.val = 0, this.left, this.right]);
 * }
 */
class Solution {
    TreeNode? sortedArrayToBST(List<int> nums) {
        return buildBST(nums, 0, nums.length - 1);
    }
    
    TreeNode? buildBST(List<int> nums, int start, int end) {
        if (start > end) return null;
        
        int mid = start + (end - start) ~/ 2;
        TreeNode node = TreeNode(nums[mid]);
        node.left = buildBST(nums, start, mid - 1);
        node.right = buildBST(nums, mid + 1, end);
        
        return node;
    }
}
